crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -941,8 +941,8 @@ fn type_expression() -> impl NoirParser<UnresolvedTypeExpression> {
     recursive(|expr| {
         expression_with_precedence(
             Precedence::lowest_type_precedence(),
-            nothing(),
             expr,
+            nothing(),
```

Comment:
Can't believe this wasn't caught earlier, thanks for finding the bug!

Reply:
@jfecher :smiley: :shipit: